google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -456,8 +456,39 @@ void ComposeObjectFromEncryptedObjects(google::cloud::storage::Client client,
               << " Metadata: " << composed_object << std::endl;
   }
   //! [compose object from encrypted objects]
-  (std::move(client), bucket_name, destination_object_name,
-   base64_aes256_key, std::move(compose_objects));
+  (std::move(client), bucket_name, destination_object_name, base64_aes256_key,
+   std::move(compose_objects));
+}
+
+void WriteObjectWithKmsKey(google::cloud::storage::Client client, int& argc,
+                           char* argv[]) {
+  if (argc < 3) {
+    throw Usage{
+        "write-object-with-kms-key <bucket-name> <object-name>"
+        " <kms-key-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto object_name = ConsumeArg(argc, argv);
+  auto kms_key_name = ConsumeArg(argc, argv);
+
+  //! [write object with kms key] [START storage_upload_with_kms_key]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string object_name,
+     std::string kms_key_name) {
+    gcs::ObjectWriteStream stream = client.WriteObject(
+        bucket_name, object_name, gcs::KmsKeyName(kms_key_name));
+
+    for (int lineno = 0; lineno != 10; ++lineno) {
+      // Add 1 to the counter, because it is conventional to number lines
+      // starting at 1.
+      stream << (lineno + 1) << ": Someday, I will write creative examples\n";
```

Comment:
This is wonderful :P

Reply:
Should probably update the string to a slightly more positive string!

`: Using GCS C++ client library to write this line`